pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -254,46 +245,97 @@ func selectNodesForPreemption(
 				Pods:             pods,
 				NumPDBViolations: int64(numPDBViolations),
 			}
-			nodeNameToVictims[potentialNodes[i].Node().Name] = &victims
+			s := strategy{
+				victims:           &victims,
+				nominatedNodeName: nodeInfoCopy.Node().Name,
+			}
+			strategies = append(strategies, &s)
 			resultLock.Unlock()
 		}
 	}
 	parallelize.Until(ctx, len(potentialNodes), checkNode)
-	return nodeNameToVictims, nil
+	return strategies
 }
 
-// processPreemptionWithExtenders processes preemption with extenders
-func processPreemptionWithExtenders(fh framework.FrameworkHandle, pod *v1.Pod, nodeNameToVictims map[string]*extenderv1.Victims) (map[string]*extenderv1.Victims, error) {
-	if len(nodeNameToVictims) > 0 {
-		for _, extender := range fh.PreemptHandle().Extenders() {
-			if extender.SupportsPreemption() && extender.IsInterested(pod) {
-				newNodeNameToVictims, err := extender.ProcessPreemption(
-					pod,
-					nodeNameToVictims,
-					fh.SnapshotSharedLister().NodeInfos(),
-				)
-				if err != nil {
-					if extender.IsIgnorable() {
-						klog.Warningf("Skipping extender %v as it returned error %v and has ignorable flag set",
-							extender, err)
-						continue
-					}
-					return nil, err
-				}
-
-				// Replace nodeNameToVictims with new result after preemption. So the
-				// rest of extenders can continue use it as parameter.
-				nodeNameToVictims = newNodeNameToVictims
+// CallExtenders call given <extenders> to filter out <strategies>, and return a filtered strategies.
+// We will only check <strategies> with extenders that support preemption.
+// Extenders which do not support preemption may later prevent preemptor from being scheduled on the nominated
+// node. In that case, scheduler will find a different host for the preemptor in subsequent scheduling cycles.
+func CallExtenders(extenders []framework.Extender, pod *v1.Pod, nodeLister framework.NodeInfoLister,
```

Comment:
perhaps it would have a more descriptive name?  for example, `CallPreemptExtenders`??

Reply:
SG.